src/components/application_manager/src/rpc_passing_handler.cc

```diff
@@ -228,6 +233,20 @@ void RPCPassingHandler::ForwardRequestToCore(uint32_t correlation_id) {
       std::make_shared<smart_objects::SmartObject>(message);
   rpc_request_queue.erase(correlation_id);
   rpc_request_queue_lock_.Release();
+
+  // Revalidate rpc message before forwarding to core
+  uint32_t connection_key =
+      message[strings::params][strings::connection_key].asUInt();
+  int32_t function_id = message[strings::params][strings::function_id].asInt();
+
+  auto app_ptr = app_manager_.application(connection_key);
+  utils::SemanticVersion msg_version(0, 0, 0);
+  if (app_ptr) {
+    msg_version = app_ptr->msg_version();
+  }
+
+  app_manager_.GetRPCHandler().ValidateRPCSO(
```

Comment:
You need to check the result of this function.

Reply:
I guess I would simply return in the case the validation fails since the function already handles sending the `INVALID_DATA` response?

Reply:
Made the change ```if (!app_manager_.GetRPCHandler().ValidateRpcSO(
          &message, msg_version, report, true)) {
    std::shared_ptr<smart_objects::SmartObject> response(
        MessageHelper::CreateNegativeResponse(
            connection_key,
            function_id,
            correlation_id,
            mobile_apis::Result::INVALID_DATA));

    (*response)[strings::msg_params][strings::info] = rpc::PrettyFormat(report);
    app_manager_.GetRPCService().ManageMobileCommand(
        response, commands::Command::SOURCE_SDL);
    return;
  }```